File: js/Sprite.js


class Sprite
{
   constructor()
   {
      this.R = new Rect(0,0,16,16)
      this.frames=[
      new Rect(43, 9, 16, 16),
       new Rect(60, 9, 16, 16),
      new Rect(77, 9, 16, 16)]
      this.frame = 0.0;
      this.animationSpeed = 10.0
      this.hidden=false
   }
   MoveFrame(frametime)
   {

      this.frame += frametime * this.animationSpeed;
      // #print(this.frame,frametime)
      if (this.frame >= this.frames.length)
      {
         this.frame = 0;
      }
   }
}
  

File: py/PyGameTutorial/Sprite.py

import pygame

class Sprite(object):
    def __init__(self):
        self.R = pygame.Rect(0,0,16,16)
        self.frames=[
                   pygame.Rect(43, 9, 16, 16),
                   pygame.Rect(60, 9, 16, 16),
                   pygame.Rect(77, 9, 16, 16)]
        self.frame = 0.0;
        self.animationSpeed = 10.0
        self.hidden=False

    def MoveFrame(self,frametime):
        
        self.frame += frametime * self.animationSpeed;
        #print(self.frame,frametime)
        if self.frame >= len(self.frames):
           self.frame = 0;
        

